sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeExtractors.scala

```diff
@@ -117,6 +122,8 @@ abstract class ExtractValueWithStruct extends UnaryExpression with ExtractValue
 
 /**
  * Returns the value of fields in the Struct `child`.
+ *
+ * No need to do type checking since it is handled by [[ExtractValue]].
```

Comment:
We already have doc for `trait ExtractValue` says "Note: concrete extract value expressions are created only by `ExtractValue.apply`, we don't need to do type check for them."  Should we say it again here?